cloudant-client/src/test/java/com/cloudant/tests/SessionInterceptorExpiryTests.java

```diff
@@ -0,0 +1,166 @@
+            mockIamServer.enqueue(new MockResponse().setResponseCode(200)
+                    .setBody(MockWebServerResources.IAM_TOKEN));
+            cookieString = MockWebServerResources.iamSessionCookie(cookieValue, expiry);
+        }
+        MockResponse cookieResponse = new MockResponse()
+                .setResponseCode(200)
+                .setBody(MockWebServerResources.MOCK_COOKIE_RESPONSE_BODY)
+                .addHeader("Set-Cookie", cookieString);
+        mockWebServer.enqueue(cookieResponse);
+
+        // Followed by an OK response
+        mockWebServer.enqueue(MockWebServerResources.JSON_OK);
+    }
+
+    private void executeTest(Long expiryTime, String cookieValue) throws Exception {
+        queueResponses(expiryTime, cookieValue);
+        HttpConnection conn = Http.GET(mockWebServer.url("/").url());
+        conn.requestInterceptors.add(rqInterceptor);
+        conn.responseInterceptors.add(rpInterceptor);
+        conn.execute();
+
+        // Assert the _session request
+        RecordedRequest sessionRequest = mockWebServer.takeRequest(MockWebServerResources
+                .TIMEOUT, MockWebServerResources.TIMEOUT_UNIT);
+
+        assertEquals("The interceptor should make a session request", sessionPath,
+                sessionRequest.getPath());
+        assertNull("There should be no existing cookie on the session request", sessionRequest
+                .getHeader("Cookie"));
+
+        // Assert the GET request
+        RecordedRequest getRequest = mockWebServer.takeRequest(MockWebServerResources.TIMEOUT,
+                MockWebServerResources.TIMEOUT_UNIT);
+        assertEquals("The request path should be correct", "/", getRequest.getPath());
+        assertNotNull("There should be a cookie on the request", getRequest.getHeader("Cookie"));
+        String expectedCookie = ((sessionPath.equals("/_session")) ? MockWebServerResources.AUTH_COOKIE_NAME :
+                MockWebServerResources.IAM_COOKIE_NAME) + "=" + cookieValue;
+        assertEquals("The cookie should be the correct session type", expectedCookie,
+                getRequest.getHeader("Cookie"));
+    }
+
+    /**
+     * Test the non-expiry case just to validate that things work normally
+     * @throws Exception
+     */
+    @Test
+    public void testMakesCookieRequest() throws Exception {
+        executeTest(null, MockWebServerResources.EXPECTED_OK_COOKIE);
+    }
+
+    /**
+     * Test that if a cookie is expired it does not cause a replay cycle. That is we should not
+     * retrieve an expired cookie from the store, so a new session request should be made before
+     * any subsequent request.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testNewCookieRequestMadeAfterExpiry() throws Exception {
+        // Make a GET request and get a cookie valid for 2 seconds
+        executeTest(System.currentTimeMillis() + 2000, MockWebServerResources.EXPECTED_OK_COOKIE);
```

Comment:
Yes, although I guess it depends what we are trying to verify since I think the expired cookie is never added to the store vs being in the store, but not being returned on the next request.